src/lib.rs

```diff
@@ -55,16 +55,32 @@
 //! [`defer`]: fn.defer.html
 
 #![cfg_attr(feature = "nightly", feature(const_fn))]
+#![cfg_attr(feature = "nightly", feature(alloc))]
+#![cfg_attr(not(test), no_std)]
+
+#[cfg(all(not(test), feature = "std"))]
+#[macro_use]
+extern crate std;
+#[cfg(test)]
+extern crate core;
+
+// Use liballoc on nightly to avoid a dependency on libstd
+#[cfg(feature = "nightly")]
+extern crate alloc;
+#[cfg(not(feature = "nightly"))]
+extern crate std as alloc;
```

Comment:
Even if we rename `std` into `alloc` here, there's another branch on `cfg(feature = "nightly")` in importing `Arc`.  If so, how about:

- just `extern crate std` here;
- import `Box` and `Arc` in `lib.rs` depending on `fg(feature = "nightly")`;
- use `epoch::Box` and `epoch::Arc` elsewhere?